### PR TITLE
Add CompleteCommand

### PR DESCRIPTION
**Pull request by jia1**

### CODE REVIEW

File: src/main/java/seedu/taskman/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -18,9 +18,9 @@
 
     public static final String COMMAND_WORD = "edit";
 
-    // kiv: let parameters be objects. we can easily generate the usage in that case
-    // todo: update message
-    // UG/DG
+    // KIV: Let parameters be objects. We can easily generate the usage in that case.
+    // TODO: Update message. I'm so confused about the order of the arguments urgh...
+    // UG/DG (what is this for?)
```
Comment (alexfjw) [inline src/main/java/seedu/taskman/logic/commands/EditCommand.java]: for the day we make the changes

File: src/main/java/seedu/taskman/logic/commands/CompleteCommand.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	
+	public static final String MESSAGE_USAGE = COMMAND_WORD + ": Marks an existing task as complete. "
+            + "Parameters: INDEX\n"
+            + "Example: " + COMMAND_WORD
+            + " 1";
+
+    public static final String MESSAGE_SUCCESS = "Task completed: %1$s";
+
+    public CompleteCommand(int targetIndex,
+            @Nullable String title, @Nullable String deadline, @Nullable String status,
```
Comment (alexfjw) [inline src/main/java/seedu/taskman/logic/commands/CompleteCommand.java]: How about using a constructor which takes only the targetIndex ?

File: src/main/java/seedu/taskman/logic/parser/CommandParser.java [hunk 3]
```diff
@@ -209,6 +209,25 @@ private Command prepareEdit(String args) {
...
+
+        Optional<Integer> index = parseIndex(args);
+        if(!index.isPresent()){
+            return new IncorrectCommand(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, CompleteCommand.MESSAGE_USAGE));
+        }
+
+        return new CompleteCommand(
+		        index.get(),
+		        null,
```
Comment (alexfjw) [inline src/main/java/seedu/taskman/logic/parser/CommandParser.java]: As above, constructor which only takes targetIndex would be more appropriate. Don't need to put the nulls.

File: src/main/java/seedu/taskman/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -98,35 +98,39 @@ private void initMembers(ArgumentContainer argsContainer) throws IllegalValueExc
...
+                argsContainer.schedule == null
+                        ? beforeEdit.getSchedule().orElse(null)
+                        : new Schedule (argsContainer.schedule),
+                argsContainer.frequency == null
+                        ? beforeEdit.getFrequency().orElse(null)
+                        : new Frequency(argsContainer.frequency)
+            	);
+            	task.status = argsContainer.status == null
+                        ? beforeEdit.getStatus().orElse(null)
+                        : new Status(argsContainer.status);
```
Comment (alexfjw) [inline src/main/java/seedu/taskman/logic/commands/EditCommand.java:119]: Its supposed to edit the task to completed here. (Not check the args passed in to the complete command)

Comment (alexfjw) [reply src/main/java/seedu/taskman/logic/commands/EditCommand.java:119]: Or maybe it should take in arguments for the 'status' field. Depends on what you have in mind for complete. 

We could either go with the twin commands complete & incomplete, 
(revise the naming as you see fit)

or
Have complete take in different arguments to support setting to completed & uncompleted.

File: src/main/java/seedu/taskman/logic/commands/EditCommand.java (hunk 1 above)
Comment (jia1) [reply src/main/java/seedu/taskman/logic/commands/EditCommand.java]: ?

Comment (alexfjw) [reply src/main/java/seedu/taskman/logic/commands/EditCommand.java]: Just remove the 3 lines. Might not make much sense anymore, given our current direction.

File: src/main/java/seedu/taskman/logic/commands/EditCommand.java (hunk 4 above)
Comment (jia1) [reply src/main/java/seedu/taskman/logic/commands/EditCommand.java:119]: This one can be easily changed. KIV.

**Comment by jia1** [comment]
OKAY CompleteCommand takes 1 arg, works, and has tests!!!! @CS2103AUG2016-W10-C1/developers

**Pull request merged**